py/server/tests/test_liveness.py

```diff
@@ -52,6 +53,8 @@ def test_liveness(self):
             l_scope.preserve(must_keep)
             l_scope.close()
             l_scope_2.close()
+        l_scope_2.close()
+        l_scope.close()
```

Comment:
to confirm, the preceding `with` should throw, so neither close() call in that block can be invoked? If so, why even have the close() calls there, if they are effectively "dead code"?

Reply:
It is these calls (out of order) that will throw. The added lines are just to correctly pop the liveness scopes so that the tearDown won't complain.